### PR TITLE
Add support for DeveloperPortal and OpenApiJson format

### PR DESCRIPTION
**Pull request by solankisamir**
## Description

* Added support for retrieving and configuring Custom Domain on the `DeveloperPortal` Endpoint
    https://github.com/Azure/azure-powershell/issues/11007

* **Export-AzApiManagementApi** Added support for downloading Api Definition in Json format
    https://github.com/Azure/azure-powershell/issues/9987

* **Import-AzApiManagementApi** Added support for importing OpenApi 3.0 definition from Json document

* **New-AzApiManagementIdentityProvider** and **Set-AzApiManagementIdentityProvider** Added support for configuring `Signin Tenant` for AAD B2C Provider
    https://github.com/Azure/azure-powershell/issues/9784

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [x] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [x] If applicable, the changes made in the PR have proper test coverage
- [ ] For public API changes to cmdlets:
    - [ ] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
    - [ ] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

File: src/ApiManagement/ApiManagement.ServiceManagement/ApiManagementClient.cs [hunk 1]
```diff
@@ -1077,7 +1078,10 @@ byte[] DownloadFileFromLink(string url)
                     headerValue = fromFile ? ContentFormat.SwaggerJson : ContentFormat.SwaggerLinkJson; 
                     break;
                 case PsApiManagementApiFormat.OpenApi:
-                    headerValue = fromFile ? ContentFormat.Openapijson : ContentFormat.OpenapiLink;
+                    headerValue = fromFile ? ContentFormat.Openapi : ContentFormat.OpenapiLink;
```
Comment (wyunchi-ms) [inline src/ApiManagement/ApiManagement.ServiceManagement/ApiManagementClient.cs]: For the case that `specificationApiFormat == PsApiManagementApiFormat.OpenApi` and `fromFile == true`. The return of `GetContentFormatForApiImport`will change from `ContentFormat.Openapijson` to `ContentFormat.Openapi`. Is this a bug fix or a new feature? If it's a new feature, will this affect the exist user script?

Comment (wyunchi-ms) [reply src/ApiManagement/ApiManagement.ServiceManagement/ApiManagementClient.cs]: @solankisamir

Comment (solankisamir) [reply src/ApiManagement/ApiManagement.ServiceManagement/ApiManagementClient.cs]: This is a bug fix. When `OpenApi` was specified, we should only have accept `YAML` files. 
For OpenApi 3.0 document in Json format, we have the format `OpenApiJson`

**Comment by solankisamir** [comment]
@wyunchi-ms can you take a look at this again

**Pull request merged**